projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/cutsomer-ticketing.ts

```diff
@@ -0,0 +1,104 @@
+
+export enum TestCategory {
+  enquiry = "Enquiry",
+  complaint = "Complaint",
+  problem = "Problem",
+}
+
+export interface TestTicketDetails {
+  subject: string;
+  message: string;
+  category: TestCategory;
+  associatedTo?: string;
+  filename?: string;
+}
+
+
+export function visitElectronicSupportTicketPage() {
+  cy.visit('/my-account/support-tickets');
+  cy.wait(1000);
+}
+
+
+export function openCreateTicketModal() {
+  cy.get('cx-customer-ticketing-list').should('exist');
+  cy.get('cx-customer-ticketing-create button').click();
+  cy.get('cx-customer-ticketing-create-dialog').should('exist');
+}
+
+export function fillTicketDetails(ticketDetails: TestTicketDetails){
+
+  // subject
+  cy.get('textarea').first().type(ticketDetails.subject);
+
+  // category dropdown
+  cy.contains('.cx-customer-ticket-label', 'Category').get('select').eq(2).select(ticketDetails.category); // TODO
+
+  // AssociatedTo dropdown
+  // cy.contains('.cx-customer-ticket-label', 'Associated To').get('select').eq(3).select('');
+
+  // Message
+  cy.get('textarea').last().type(ticketDetails.message);
+
+  // validation
+  cy.get('cx-form-errors').should('not.be.visible');
+}
+
+export function addFile(filename: string){
+  cy.get('cx-file-upload input[type="file"]').attachFile({filePath: '../helpers/customer-ticketing/files-to-upload/' + filename});
+  cy.get('p').contains(filename);
+}
+
+export function verifyFileAttachedToMessage(filename: string){
+  cy.get('cx-messaging').contains(filename);
+}
+
+export function clickSubmit(){
+  cy.contains('[type="button"]', 'Submit').click();
+}
+
+export function clickSubmitAndVerifyRequestCompleted(){
+  cy.contains('[type="button"]', 'Submit').click();
+  cy.get('cx-global-message').contains('Request Created');
+}
```

Comment:
You already have clickSubmit lines above no need to repeat.

```suggestion
export function verifyTicketCreationRequestCompleted(){
  cy.get('cx-global-message').contains('Request Created');
}
```

Reply:
yes, I should remove this one